docs/api.rst

```diff
@@ -74,6 +74,8 @@ Configures global options:
 * ``useBR``: a flag to generate ``<br>`` tags instead of new-line characters in the output, useful when code is marked up using a non-``<pre>`` container.
 * ``classPrefix``: a string prefix added before class names in the generated markup, used for backwards compatibility with stylesheets.
 * ``languages``: an array of language names and aliases restricting auto detection to only these languages.
+* ``languagePrefixRe``: allows you to configure the css language prefix  (to allow class names like, `color-as-php`, etc.)
```

Comment:
What about `languageNameRe` to make it clear that it doesn't actually have to be a prefix?

Reply:
I'm open to a better name. :-)  Should we just drop the prefix part?  Or just remove the body from the regex and truly make it a prefix?  I'm not sure.  Do you have a preference or thoughts?

Reply:
`languageDetectRe`?

Reply:
Hmm what's the purpose of this variable? I don't quite understand the use case of `color-as-php`, like when would you use that?

Reply:
In class on the client. 

code class="lang-php" 

Vs 

Code class="color-as-php"

Reply:
In class on the client. 

code class="lang-php" 

Vs 

Code class="color-as-php"

Reply:
So this configuration option will allow highlight.js to look for custom prefixes? So instead of looking for `lang-*` then it'll look for `color-as-*`?

Reply:
Or both. However it's configured. It's regex.

Reply:
@allejo: Just so. I don't think someone will ever change it, but who knows...
@yyyc514: I actually proposed `languageNameRe` above, but `languageDetectRe` is fine.

Reply:
How did I miss that? LOL.

Reply:
I dislike languageName because it's not really the name, it's prefix + name... so I think `name` mistates it a bit.  I'll change it to Detect if you're cool with that.

Reply:
Agreed.